### ISSUE TITLE
Connect failed when using Browser

### ISSUE THREAD

**Issue by LangYuanzh** [neo4j/neo4j#13174]
- Neo4j version: 5.5.0 comunity
- Operating system: Centos7

It takes very long time when connect to neo4j DBMS through neo4j browser with bolt. And the WebSocket always connect failed. I find some exception in the debug.log of neo4j:

2023-04-25 02:00:15.229+0000 ERROR [o.n.b.p.c.h.ProtocolHandshakeHandler] Fatal error occurred during protocol handshaking: [id: 0xb03bd313, L:/10.65.18.40:7687 - R:/10.65.20.62:62516]
java.lang.NullPointerException: Cannot read the array length because "array" is null
        at io.netty.buffer.Unpooled.copiedBuffer(Unpooled.java:363) ~[netty-buffer-4.1.87.Final.jar:4.1.87.Final]
        at org.neo4j.bolt.protocol.common.handler.DiscoveryResponseHandler.channelRead(DiscoveryResponseHandler.java:58) ~[neo4j-bolt-5.5.0.jar:5.5.0]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.handlerRemoved(ByteToMessageDecoder.java:266) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:536) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290) ~[netty-codec-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:397) ~[netty-transport-classes-epoll-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.87.Final.jar:4.1.87.Final]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by jotamorais** [comment]
Experiencing the same on `Rocky Linux 8.7`

Server specs:
```
50Gi Disk
16Gi memory
Intel(R) Xeon(R) CPU @ 2.20GHz 
```
Running on Docker container with image `neo4j:5.8-community@sha256:3546a3130e80a51d420e7a07e925aa6465d2af87b77add87cb35f43c7a14caeb` .

Here are my settings:

```Properties
# Raise memory limits
NEO4J_server_memory_heap_initial__size=5g
NEO4J_server_memory_heap_max__size=5g
NEO4J_server_memory_pagecache_size=7g
NEO4J_db_memory_transaction_total_max=0

# DB tunning
NEO4J_db_transaction_timeout=0

# Authentication
NEO4J_AUTH=[Redacted]/[Redacted]
# NEO4J_AUTH=none

# It is also recommended turning out-of-memory errors into full crashes,
# instead of allowing a partially crashed database to continue running:
NEO4J_server_jvm_additional='-XX:+ExitOnOutOfMemoryError'

NEO4J_server_config_strict__validation_enabled=true
NEO4J_dbms_security_procedures_allowlist=apoc.coll.*,apoc.load.*,apoc.*
NEO4J_dbms_security_procedures_unrestricted=apoc.*,algo.*

# https://neo4j.com/docs/upgrade-migration-guide/current/version-5/migration/breaking-changes/#_apoc_core
apoc.export.file.enabled=true
apoc.import.file.enabled=true

# SSL Settings
# https://neo4j.com/docs/operations-manual/current/security/ssl-framework/
NEO4J_server_https_enabled=true
NEO4J_server_default__listen__address=0.0.0.0
# HTTP Connector. There can be zero or one HTTP connectors.
NEO4J_server_https_listen__address=0.0.0.0:7473
NEO4J_server_http_enabled=false
NEO4J_dbms_ssl_policy_https_client__auth=OPTIONAL
NEO4J_dbms_ssl_policy_https_enabled=true
NEO4J_dbms_ssl_policy_https_base__directory=/var/lib/neo4j/certificates/https
NEO4J_dbms_ssl_policy_https_private__key=/var/lib/neo4j/certificates/private.key
NEO4J_dbms_ssl_policy_https_public__certificate=/var/lib/neo4j/certificates/https/trusted/public.crt

# Bolt connector
NEO4J_server_bolt_enabled=true
NEO4J_dbms_ssl_policy_bolt_client__auth=OPTIONAL
NEO4J_server_bolt_tls__level=REQUIRED
NEO4J_server_bolt_listen__address=0.0.0.0:7687
NEO4J_dbms_ssl_policy_bolt_enabled=true
NEO4J_dbms_ssl_policy_bolt_base__directory=/var/lib/neo4j/certificates/bolt
NEO4J_dbms_ssl_policy_bolt_private__key=/var/lib/neo4j/certificates/private.key
NEO4J_dbms_ssl_policy_bolt_public__certificate=/var/lib/neo4j/certificates/bolt/trusted/public.crt
```

**Comment by gjmwoods** [comment]
@LangYuanzh @jotamorais Many thanks for raising this issue. This was a bug which we have fixed in 5.10 ([this is the fix commit](https://github.com/neo4j/neo4j/commit/139d5612c2abdf15df68cdb6b43812472461cad5)).
